### PR TITLE
Add acceptance tests around Investment Project Evaluations (includes bug fix)

### PR DESCRIPTION
**Pull request by yeahfro**
DH-1279

This will pre-empt some investment project work coming in the next sprint. We currently do not have any coverage around the Evaluations page for investment projects.

A bug has also been fixed as part of this change. The bug is around the display of `New jobs` and `Safeguarded jobs` when the value is `0`.

# Before
screen shot 2017-12-29 at 11 38 35

# After
screen shot 2017-12-29 at 11 37 56

### CODE REVIEW

File: config/nunjucks/filters.js [hunk 1]
```diff
@@ -113,14 +113,14 @@ const filters = {
 
     if (isArray(collection)) {
       return collection.slice().map((element) => {
-        return !element || isEmptyObjOrArray(element) ? defaultValue : element
+        return isNil(element) || element === '' || isEmptyObjOrArray(element) ? defaultValue : element
```
Comment (ztolley) [inline config/nunjucks/filters.js]: I assumed isEmpty would do this but let me check.

File: test/acceptance/features/investment-projects/evaluations-details.feature [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    Given I navigate to company fixture Lambda plc
+    When I click the Investment local nav link
+    And I click the "Add investment project" link
+    Then I am taken to the "Add investment project" page
+    When I select FDI as the Investment project type
+    And I choose Yes for "Will this company be the source of foreign equity investment?"
+    And I populate the create Investment Project form
+    Then I see the success message
+    When I click the "Add value" link
+    And I populate the create investment project value form all Yes
```
Comment (ztolley) [inline test/acceptance/features/investment-projects/evaluations-details.feature]: 'all yes' ?

File: test/acceptance/features/investment-projects/evaluations-details.feature [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      | key                        | value  |
+      | Number of new jobs         | 100    |
+      | Number of safeguarded jobs | 200    |
+      | Total investment           | 100000 |
+      | Foreign equity investment  | 200000 |
+    Then I see the success message
+    When I click the Evaluations local nav link
+    Then the FDI (Test A) details are displayed
+      | key                               | value                |
```
Comment (ztolley) [inline test/acceptance/features/investment-projects/evaluations-details.feature]: Deffo like this and the use of parameter data above, we need to use this to test editing things as well as creating.

File: test/acceptance/features/investment-projects/evaluations-details.feature [hunk 4]
```diff
@@ -14,24 +14,35 @@ Feature: Investment projects evaluations details
     Then I see the success message
     When I click the "Add value" link
     And I populate the create investment project value form all Yes
-      | key                        | value  |
-      | Number of new jobs         | 100    |
-      | Number of safeguarded jobs | 200    |
-      | Total investment           | 100000 |
-      | Foreign equity investment  | 200000 |
+      | key                               | value                                          |
```
Comment (ztolley) [inline test/acceptance/features/investment-projects/evaluations-details.feature]: Why was this added in one commit and then edited in the next?

File: config/nunjucks/filters.js (hunk 1 above)
Comment (ztolley) [reply config/nunjucks/filters.js]: Yup use isEmpty, it checks all of these

File: config/nunjucks/filters.js [hunk 5]
```diff
@@ -107,20 +109,18 @@ const filters = {
   },
 
   collectionDefault: (collection, defaultValue = 'Not found') => {
-    const isEmptyObjOrArray = (value) => {
-      return ((isPlainObject(value) || isArray(value)) && isEmpty(value))
-    }
+    const hasValue = (element) => !isEmpty(element) || isBoolean(element) || isInteger(element)
```
Comment (feedmypixel) [inline config/nunjucks/filters.js]: I'd stick this on a separate line and add braces, its currently a little difficult to read

File: config/nunjucks/filters.js [hunk 6]
```diff
@@ -107,20 +109,18 @@ const filters = {
...
     if (isArray(collection)) {
       return collection.slice().map((element) => {
-        return !element || isEmptyObjOrArray(element) ? defaultValue : element
+        return !hasValue(element) ? defaultValue : element
       })
     }
     if (isPlainObject(collection)) {
       const obj = Object.assign({}, collection)
 
       Object.keys(obj).forEach((key) => {
```
Comment (feedmypixel) [inline config/nunjucks/filters.js]: can we use `keys` here? in the same vein as we have used `assign` over `Object.assign`

Comment (feedmypixel) [reply config/nunjucks/filters.js]: and maybe `lodash` `forEach` whilst your there?

File: config/nunjucks/filters.js (hunk 5 above)
Comment (feedmypixel) [inline config/nunjucks/filters.js]: not sure `hasValue` works when its read in the code, but I can't think of anything better :)

something like:
```
const nillOrEmpty = (value) => {
 return (isNull(value) && isEmpty(value))
}
```

`!nillOrEmpty(element) ? defaultValue : element`

File: test/acceptance/features/details/step_definitions/details.js [hunk 7]
```diff
@@ -4,6 +4,12 @@ const { get, includes } = require('lodash')
 
 const { getDetailsTableRowValue } = require('../../../helpers/selectors')
 
+const formatters = {
```
Comment (feedmypixel) [inline test/acceptance/features/details/step_definitions/details.js]: I wonder if these can be abstracted out in a future PR when used by others?

File: test/acceptance/features/investment-projects/evaluations-details.feature (hunk 2 above)
Comment (ztolley) [reply test/acceptance/features/investment-projects/evaluations-details.feature]: As mentioned, feels a bit like the function has a dual personality if you say set all things to yes AND pass in params. I'd take the 'all yes' bit off and set the yes values in the parameter table.

File: config/nunjucks/filters.js (hunk 5 above)
Comment (yeahfro) [reply config/nunjucks/filters.js]: Personally I don't find these inline functions hard to read but for the sake of adding a couple of curly braces I have done that.

Comment (yeahfro) [reply config/nunjucks/filters.js]: Your function does not work because `isEmpty` returns true for `true`, `false`, `0`, `1` etc. I have left the `isBoolean` and `isInteger` but renamed the function to `hasPresentableValue`
